cpp/src/table/legacy/device_table_row_operators.cuh

```diff
@@ -187,6 +224,46 @@ struct typed_row_inequality_comparator
         return State::False;
     }
   }
+  template <typename ColType>
+  __device__
+    typename std::enable_if_t<std::is_floating_point<ColType>::value, State>
+      operator()(gdf_index_type lhs_row, gdf_index_type rhs_row,
+                 gdf_column const *lhs_column, gdf_column const *rhs_column,
```

Comment:
Why does this operator take pointers to columns and the other ones take references?